pkg/doc/verifiable/credential.go

```diff
@@ -369,26 +389,26 @@ type Credential struct {
 	Schemas        []TypedID
 	Evidence       *Evidence
 	TermsOfUse     []TypedID
-	RefreshService *TypedID
+	RefreshService []TypedID
 
 	CustomFields CustomFields
 }
 
 // rawCredential is a basic verifiable credential
 type rawCredential struct {
-	Context        interface{} `json:"@context,omitempty"`
-	ID             string      `json:"id,omitempty"`
-	Type           interface{} `json:"type,omitempty"`
-	Subject        Subject     `json:"credentialSubject,omitempty"`
-	Issued         *time.Time  `json:"issuanceDate,omitempty"`
-	Expired        *time.Time  `json:"expirationDate,omitempty"`
-	Proof          *Proof      `json:"proof,omitempty"`
-	Status         *TypedID    `json:"credentialStatus,omitempty"`
-	Issuer         interface{} `json:"issuer,omitempty"`
-	Schema         interface{} `json:"credentialSchema,omitempty"`
-	Evidence       *Evidence   `json:"evidence,omitempty"`
-	TermsOfUse     []TypedID   `json:"termsOfUse,omitempty"`
-	RefreshService *TypedID    `json:"refreshService,omitempty"`
+	Context        interface{}     `json:"@context,omitempty"`
+	ID             string          `json:"id,omitempty"`
+	Type           interface{}     `json:"type,omitempty"`
+	Subject        Subject         `json:"credentialSubject,omitempty"`
+	Issued         *time.Time      `json:"issuanceDate,omitempty"`
+	Expired        *time.Time      `json:"expirationDate,omitempty"`
+	Proof          *Proof          `json:"proof,omitempty"`
+	Status         *TypedID        `json:"credentialStatus,omitempty"`
+	Issuer         interface{}     `json:"issuer,omitempty"`
+	Schema         interface{}     `json:"credentialSchema,omitempty"`
+	Evidence       *Evidence       `json:"evidence,omitempty"`
+	TermsOfUse     json.RawMessage `json:"termsOfUse,omitempty"`
```

Comment:
why was the change from TypedID to RawMessage needed?

Reply:
Because we don't know is it defined as a single object or an array. That's why we postpone the parsing.